src/main/java/org/jabref/logic/pdf/PdfAnnotationImporter.java

```diff
@@ -39,115 +44,123 @@
-
-                            k += 8;
-
-                            float ulx = upperLeftX.floatValue() - 1;
-                            float uly = upperLeftY.floatValue();
-                            float width = upperRightX.floatValue() - lowerLeftX.floatValue();
-                            float height = upperRightY.floatValue() - lowerLeftY.floatValue();
-
-                            PDRectangle pageSize = page.getMediaBox();
-                            uly = pageSize.getHeight() - uly;
-
-                            Rectangle2D.Float rectangle = new Rectangle2D.Float(ulx, uly, width, height);
-                            stripperByArea.addRegion("highlightedRegion", rectangle);
-                            stripperByArea.extractRegions(page);
-                            String highlightedTextInLine = stripperByArea.getTextForRegion("highlightedRegion");
-
-                            if (j > 1) {
-                                highlightedText = highlightedText.concat(highlightedTextInLine);
-                            } else {
-                                highlightedText = highlightedTextInLine;
-                            }
-                        }
-                        annotation.setContents(highlightedText);
-
-                        FileAnnotation highlighting = new FileAnnotation(annotation, i + 1);
-                        //highlighted text that has a sticky note on it should be linked to the sticky note
-                        highlighting.linkComments(annotationBelongingToHighlighting);
-                        annotationsList.add(annotationBelongingToHighlighting);
-                        annotationsList.add(highlighting);
+        try {
+            annotation.setContents(extractHighlightedText(page, annotation));
+        } catch (IOException e) {
+            annotation.setContents("JabRef: Could not extract any highlighted text!");
+        }
 
-                    } else {
-                        annotationsList.add(new FileAnnotation(annotation, i + 1));
-                    }
-                }
-            } catch (IOException e) {
-                LOGGER.error(String.format("Failed to read file %s.", path) , e);
+        //highlighted text that has a sticky note on it should be linked to the sticky note
+        highlightAnnotations.add(new FileAnnotation(annotation, pageIndex + 1, annotationBelongingToHighlighting));
+        return highlightAnnotations;
+    }
+
+    private String extractHighlightedText(PDPage page, PDAnnotation annotation) throws IOException {
+        //highlighted text has to be extracted by the rectangle calculated from the highlighting
+        PDFTextStripperByArea stripperByArea = new PDFTextStripperByArea();
+        COSArray quadsArray = (COSArray) annotation.getDictionary().getDictionaryObject(COSName.getPDFName("QuadPoints"));
+        String highlightedText = "";
+        for (int j = 1,
```

Comment:
I don't know if you changed it, but it would be nice to have a bit better speaking variable names, as I here don't get from a quick view what j and k stand for.